### ISSUE TITLE
[Table] [A11Y] Table - Placeholder text in table cell header inputs are not readable

### ISSUE THREAD

**Issue by JordanWSmith15** [carbon-design-system/carbon-addons-iot-react#2788]
Split from https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2620

Describe the bug

===== Accessibility checkpoint =====
Test Tool Type: IBM Equal Access Accessibility Checker
Checkpoint: IBM 2.1.1 Keyboard (https://www.ibm.com/able/requirements/requirements/#2_1_1)

=====STEPS TO REPRODUCE =====

Open file storage (https://cloud.ibm.com/cloud-storage/file) or block storage (https://cloud.ibm.com/cloud-storage/block)
Login with ID & password
Use Tab key to navigate on the Filter button, tap Enter or Space to display the filter row
Continue use Tab key to navigate on location combobox
Tap Down arrow key to display the menu options
===== Problem Summary =====
When we use Tab key to navigate on location combobox, we cannot see the placeholder strings.

===== EXPECTED RESULT(Accessibility guidance) =====
TBD. See discussion in https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2620
